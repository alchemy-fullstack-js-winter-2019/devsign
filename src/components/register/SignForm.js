import React from 'react';
import PropTypes from 'prop-types';

function SignForm({ signType, username, password, onChange, onSubmit }) {
  return (
    <>
      <h2>{signType}:</h2>
      <form onSubmit={onSubmit}>
        <label>
          <p>Username: <input name="username" type="text" value={username} onChange={onChange} /></p>
        </label>
        <label>
          <p>Password: <input name="password" type="password" value={password} onChange={onChange} /></p>
        </label>
        <button type="submit">{signType}</button>
      </form>
    </>
  );
}

SignForm.propTypes = {
  signType: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func
};

export default SignForm;
