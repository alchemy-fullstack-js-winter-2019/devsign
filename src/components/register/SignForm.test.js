import React from 'react';
import renderer from 'react-test-renderer';
import SignForm from './SignForm';

describe('Sign Form', () => {
  it('displays registration form', () => {
    const props = {
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      username: 'user123',
      password: 'passittotheword',
      signType: 'Sign In'
    };

    const tree = renderer.create(
      <SignForm {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
