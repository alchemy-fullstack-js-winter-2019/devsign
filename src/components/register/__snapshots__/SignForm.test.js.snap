// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Form displays registration form 1`] = `
Array [
  <h2>
    Sign In
    :
  </h2>,
  <form
    onSubmit={[MockFunction]}
  >
    <label>
      <p>
        Username: 
        <input
          name="username"
          onChange={[MockFunction]}
          type="text"
          value="user123"
        />
      </p>
    </label>
    <label>
      <p>
        Password: 
        <input
          name="password"
          onChange={[MockFunction]}
          type="password"
          value="passittotheword"
        />
      </p>
    </label>
    <button
      type="submit"
    >
      Sign In
    </button>
  </form>,
]
`;
